sconfig/shared/src/main/scala/org/ekrich/config/impl/SimpleConfigObject.scala

```diff
@@ -584,7 +581,21 @@ final class SimpleConfigObject(
                   newLastChar // should extend path only on identifier
                 )) sb.append('.')
           }
-          printCommentsToBuffer(sb, options, indentVal, aggComments)
+          if (atRoot)
+            printCommentsToBuffer(sb, options, indentVal, aggComments)
+          else if (!atRoot && origin._lineNumber == leafValue.origin.lineNumber) {
+            // another NASTY change sb history
+            def appendAfterLastNewLine(at: jl.StringBuilder) = {
+              // if at first line(no \n) then returns -1, -1 + 1 = 0 which is a valid idx
+              val lastNew: Int = at.lastIndexOf("\n")
+              val indent =
+                at.substring(lastNew + 1).takeWhile(_.isWhitespace).length
+              val tmp = new jl.StringBuilder()
```

Comment:
Just a comment here like we discussed on chat will be helpful for future maintainers.